core/src/main/kotlin/com/strumenta/kolasu/transformation/DummyNodes.kt

```diff
@@ -0,0 +1,96 @@
+package com.strumenta.kolasu.transformation
+
+import com.strumenta.kolasu.model.Node
+import com.strumenta.kolasu.model.PossiblyNamed
+import com.strumenta.kolasu.model.ReferenceByName
+import java.lang.reflect.ParameterizedType
+import kotlin.random.Random
+import kotlin.reflect.KClass
+import kotlin.reflect.KParameter
+import kotlin.reflect.full.createType
+import kotlin.reflect.full.isSubclassOf
+import kotlin.reflect.full.primaryConstructor
+import kotlin.reflect.jvm.javaType
+
+/**
+ * This logic instantiate a node of the given class with dummy values.
+ * This is useful because it permits to add an element that "fit" and make the typesystem happy.
+ * Typically, the only goal of the element would be to hold some annotation that indicates that the element
+ * is representing an error or a missing transformation or something of that sort.
+ */
+fun <T : Node> KClass<T>.dummyInstance(): T {
+    val kClassToInstantiate = this.toInstantiableType()
+    val emptyConstructor = kClassToInstantiate.constructors.find { it.parameters.isEmpty() }
+    if (emptyConstructor != null) {
+        return emptyConstructor.call()
+    }
+    val constructor = kClassToInstantiate.primaryConstructor!!
+    val params = mutableMapOf<KParameter, Any?>()
+    constructor.parameters.forEach { param ->
+        val mt = param.type.javaType
+        val value = when {
+            param.type.isMarkedNullable -> null
+            mt is ParameterizedType && mt.rawType == List::class.java -> mutableListOf<Any>()
+            (param.type.classifier as KClass<*>).isSubclassOf(Node::class) ->
+                (param.type.classifier as KClass<out Node>).dummyInstance()
+            param.type == String::class.createType() -> "DUMMY"
+            param.type.classifier == ReferenceByName::class -> ReferenceByName<PossiblyNamed>("UNKNOWN")
```

Comment:
This is unlikely, but the reference could be "resolved" to a real element called "UNKNOWN". Is it possible to have a _ReferenceByName_ subclass that can't possibly be resolved?

Reply:
I see your point, but I would aim to solve it by never trying to resolve references belonging to "dummy nodes" or "dummy trees". So I would consider:
1. Adding a flag `val Node.isDummy` that looks at the origin parameter and possibly also to the parent
2. Adding a flag in `walk` and similar methods, to give the option to skip dummy nodes
3. When resolving references use the `walk` method using the flag that skip dummy nodes
In this way we never attempt to solve references. Alternatively we can also create subclass of ReferenceByName that explode if you attempt to mark it as solved. 
What do you think?

Reply:
I'm not a fan of modifying _walk_, but the symbol resolver could apply a filter to the nodes it walks to exclude dummy ones, it makes sense